src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -1249,6 +1249,16 @@ private long EnforceConfigurationFor(DocumentsOperationContext context, string i
                     changeVector, lastModifiedTicks);
 
                 var currentRevisionsCount = GetRevisionsCount(context, id);
+
+                if (currentRevisionsCount == 0)
+                {
+                    var tombstoneTable = context.Transaction.InnerTransaction.OpenTable(TombstonesSchema, collectionName.GetTableName(CollectionTableType.Tombstones));
```

Comment:
Please use `GetDocumentOrTombstone` in the same code path where we do the document check